plugins/modules/firewalld_info.py

```diff
@@ -0,0 +1,392 @@
+    if not HAS_FIREWALLD:
+        module.fail_json(msg='python-firewall is required for the firewalld_info module.')
+
+    # If you want to show warning messages in the task running process,
+    # you can append the message to the 'warn' list.
+    warn = list()
+
+    try:
+        client = fw_client.FirewallClient()
+
+        # Gather general information of firewalld.
+        firewalld_info['version'] = get_version()
+        firewalld_info['default_zone'] = get_default_zone(client)
+
+        # Gather information for zones.
+        zones_info = dict()
+        collect_zones = list()
+        ignore_zones = list()
+        if module.params['active_zones']:
+            collect_zones = get_active_zones(client)
+        elif module.params['zones']:
+            all_zones = get_all_zones(client)
+            specified_zones = module.params['zones']
+            collect_zones = list(set(specified_zones) & set(all_zones))
+            ignore_zones = list(set(specified_zones) - set(collect_zones))
+            warn.append(
+                'Please note: zone:(%s) have been ignored in the gathering process.' % ','.join(ignore_zones))
+        else:
+            collect_zones = get_all_zones(client)
+
+        for zone in collect_zones:
+            # Gather settings for each zone based on the output of
+            # 'firewall-cmd --info-zone=<ZONE>' command.
+            zone_info = dict()
+            zone_settings = get_zone_settings(client, zone)
+            zone_info['target'] = get_zone_target(zone_settings)
+            zone_info['icmp_block_inversion'] = get_zone_icmp_block_inversion(zone_settings)
+            zone_info['interfaces'] = get_zone_interfaces(zone_settings)
+            zone_info['sources'] = get_zone_sources(zone_settings)
+            zone_info['services'] = get_zone_services(zone_settings)
+            zone_info['ports'] = get_zone_ports(zone_settings)
+            zone_info['protocols'] = get_zone_protocols(zone_settings)
+            zone_info['masquerade'] = get_zone_masquerade(zone_settings)
+            zone_info['forward_ports'] = get_zone_forward_ports(zone_settings)
+            zone_info['source_ports'] = get_zone_source_ports(zone_settings)
+            zone_info['icmp_blocks'] = get_zone_icmp_blocks(zone_settings)
+            zone_info['rich_rules'] = get_zone_rich_rules(zone_settings)
+
+            # The 'forward' parameter supports on python-firewall 0.9.0(or later).
+            if StrictVersion(firewalld_info['version']) >= StrictVersion('0.9.0'):
+                zone_info['forward'] = get_zone_forward(zone_settings)
+
+            zones_info[zone] = zone_info
+        firewalld_info['zones'] = zones_info
+    except AttributeError as e:
+        module.fail_json(msg=('firewalld probably not be running, Or the following method '
+                              'is not supported with your python-firewall version. (Error: %s)') % e)
```

Comment:
```suggestion
                              'is not supported with your python-firewall version. (Error: %s)') % to_native(e))
```

Reply:
Changed to wrap with to_native(). Thanks!